code/game/objects/items/mines/_mine.dm

```diff
@@ -183,25 +183,13 @@
 
 // This tells AI mobs to not be dumb and step on mines willingly.
 /obj/item/mine/is_safe_to_step(mob/living/L)
-	var/safe = FALSE
-
 	if(!armed)
-		safe = TRUE
-
-	if(!safe)	// If we're safe from having the flying component, skip to just returning.
-		if(!isliving(L))	// IF we're not alive, check if we're a mecha
-			if(ismecha(L))
-				var/obj/mecha/M = L
-				if(M.flying)
-					safe = TRUE
-				for(var/obj/item/mecha_parts/mecha_equipment/Mequip in M.equipment)
-					if(safe)
-						break
-					safe = Mequip.check_hover()
-			else
-				safe = TRUE
-
-		else if(L.hovering)	// If we're alive, check if we're otherwise hovering.
-			safe = TRUE
-
-	return safe
+		return TRUE
+
+	if(isliving(L) && L.hovering)
+		return TRUE
+	else if(ismecha(L))
+		var/obj/mecha/M = L
+		return M.in_flight()
+	else
+		return TRUE
```

Comment:
The default `return FALSE` is now implicit, but should this else be here?

Reply:
That else makes items not detonate mines.

Reply:
Does it detonate when a living, not-hovering mob steps on it? Because that would fail the first if, and thus land at this else.

Reply:
Ah, see you fixed that concern.